lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -66,6 +66,34 @@ class AwsCompileFunctions {
     }
   }
 
+  downloadS3Artifact(functionName) {
+    const functionObject = this.serverless.service.getFunction(functionName);
+    const artifactFilePath = _.get(functionObject, 'package.artifact') ||
+      _.get(this, 'serverless.service.package.artifact');
+
+    const regex = new RegExp('.*s3.amazonaws.com/(.+)/(.+)');
```

Comment:
Why do we support only S3 urls? 

Technically we could be more generic, and support .zip files from any remote location (?)

Reply:
Yes, good call. Initially I planned to add support for the S3 `Bucket` and `Key` properties as `package.artifact` keys as proposed in https://github.com/serverless/serverless/issues/5315.

However replacing the functions CloudFormation properties with such values opens up a can of worms because of some dependencies on the actual service / function `.zip` file (e.g. to create the function version hash). Because of that I decided to add support for an S3 object URL which happens to be the zip path.

Let's bring this up in an upcoming planning session and after that we can update the code to accommodate different artifact source.